R/subtr_baseline.R

```diff
@@ -49,44 +49,43 @@
 #' \emph{Applied Spectroscopy}, \strong{61}(11), 1225–1232.
 #' \doi{10.1366/000370207782597003}.
 #'
-#' @importFrom dplyr %>%
 #' @importFrom stats terms model.frame sd lm poly approx
+#' @importFrom data.table .SD
 #' @export
-subtr_bg <- function(x, ...) {
-  UseMethod("subtr_bg")
+subtr_baseline <- function(x, ...) {
+  UseMethod("subtr_baseline")
 }
 
-#' @rdname subtr_bg
+#' @rdname subtr_baseline
 #'
 #' @export
-subtr_bg.formula <- function(formula, data = NULL, ...) {
-  if (missing(formula) || (length(formula) != 3L) ||
-      (length(attr(terms(formula[-2L]), "term.labels")) != 1L))
-    stop("'formula' missing or incorrect")
-
-  mf <- model.frame(formula, data)
-  lst <- as.list(mf)
-  names(lst) <- c("y", "x")
-
-  do.call("subtr_bg", c(lst, list(...)))
+subtr_baseline.default <- function(x, ...) {
+  stop("object 'x' needs to be of class 'OpenSpecy'", call. = F)
 }
 
-#' @rdname subtr_bg
+#' @rdname subtr_baseline
 #'
 #' @export
-subtr_bg.data.frame <- function(x, ...) {
-  if (!all(c("wavenumber", "intensity") %in% names(x)))
-    stop("'data' must contain 2 columns named 'wavenumber' and 'intensity'")
-
-  do.call("subtr_bg", list(x$wavenumber, x$intensity, ...))
+subtr_baseline.OpenSpecy <- function(x, type = "polynomial",
+                                     degree = 8, raw = FALSE,
+                                     baseline_wavenumber, baseline_intensity,
+                                     make_rel = TRUE, ...) {
```

Comment:
What do you think about supplying the baseline as OpenSpecy object instead of individual wavenumber and intensity values? Might make the input a bit easier .. or would it create any problems?

Reply:
I like that idea! It will break the current shiny app implementation but it's easy to work around.